component/monitoring.libsonnet

```diff
@@ -96,9 +97,37 @@ local grafana_dashboard =
     },
   };
 
+local service_monitor_metrics =
+  if std.member(inv.applications, 'prometheus') then
+    prometheus.Enable(serviceMonitor('argocd-metrics'))
+  else
+    serviceMonitor('argocd-metrics')
+;
+
+local service_monitor_server_metrics =
+  if std.member(inv.applications, 'prometheus') then
+    prometheus.Enable(serviceMonitor('argocd-server-metrics'))
+  else
+    serviceMonitor('argocd-server-metrics')
+;
+
+local service_monitor_repo_server =
+  if std.member(inv.applications, 'prometheus') then
+    prometheus.Enable(serviceMonitor('argocd-repo-server'))
+  else
+    serviceMonitor('argocd-repo-server')
+;
+
+local final_alert_rules =
+  if std.member(inv.applications, 'prometheus') then
+    prometheus.Enable(alert_rules)
+  else
+    alert_rules
+;
+
 [
-  serviceMonitor('argocd-metrics'),
-  serviceMonitor('argocd-server-metrics'),
-  serviceMonitor('argocd-repo-server'),
-  alert_rules,
+  service_monitor_metrics,
+  service_monitor_server_metrics,
+  service_monitor_repo_server,
+  final_alert_rules,
```

Comment:
There is probably a better way to do this?

Reply:
```jsonnet
local promEnable = function(obj) 
  if std.member(inv.applications, 'prometheus') then
    prometheus.Enable(obj)
  else
    obj;

[
  promEnable(serviceMonitor('argocd-metrics')),
  ...
]
```